model/generate.go

```diff
@@ -215,8 +215,20 @@ func cacheProjectData(p *Project) projectMaps {
 
 // saveNewBuildsAndTasks saves new builds and tasks to the db.
 func (g *GeneratedProject) saveNewBuildsAndTasks(ctx context.Context, cachedProject *projectMaps, v *Version, p *Project, parentPriority int64) error {
-	newTVPairsForExistingVariants := TaskVariantPairs{}
-	newTVPairsForNewVariants := TaskVariantPairs{}
+	// inherit priority from the parent task
+	for i, projBv := range p.BuildVariants {
+		for j := range projBv.Tasks {
+			p.BuildVariants[i].Tasks[j].Priority = parentPriority
+		}
+	}
+
+	newTVPairs := TaskVariantPairs{}
+	for _, bv := range g.BuildVariants {
+		newTVPairs = appendTasks(newTVPairs, bv, p)
+	}
+	newTVPairs.ExecTasks = append(newTVPairs.ExecTasks, IncludePatchDependencies(p, newTVPairs.ExecTasks)...)
```

Comment:
Does this need to handle an execution task depending on another execution task that is already defined (so that there is a duplicate in the list)?

Reply:
Believe it or not I think I had a dream about this last night 😴 .
This is the way it worked previously, but I think this is a more general issue: `IncludePatchDependencies` __should__ return a slice that contains all the original TVPairs passed in. On the bright side, we handle duplicates when we create the tasks/builds.

This sent me down a rabbit hole: 
`IncludePatchDependencies` should return a slice that contains the original TVPairs passed in, so it should be okay to replace `newTVPairs` with the return value of `IncludePatchDependencies`. However, in one of the tests it was returning some of the original tasks and not others 😠, so the final list of tasks didn't include all of the original tasks. The reason turned out to be that a dependency of the missing task was defined as * (`AllVariants`) and the project parser turned that into a slice of every combination of the depended on task and variant, even for those variants that didn't have the task. So when we went to check on the dependencies we found it had some which didn't exist and excluded it. In the current implementation this would cause an issue for second+ degree dependencies.

My fix was to special case `AllVariants` in the project parser and not parse it, since we handle the * case downstream.